lib/ProductOpener/Ingredients.pm

```diff
@@ -1831,33 +2055,81 @@ sub parse_ingredients_text ($product_ref) {
 			# check if the whole ingredient is an ingredient
 			my $canon_ingredient = canonicalize_taxonomy_tag($ingredients_lc, "ingredients", $before);
 
-			$debug_ingredients and $log->debug(
-				"ingredient contains 'and', checking if it exists",
-				{before => $before, canon_ingredient => $canon_ingredient}
-			) if $log->is_debug();
-
 			if (not exists_taxonomy_tag("ingredients", $canon_ingredient)) {
 
+				$debug_ingredients
+					and $log->debug(
+					"parse_ingredient_text - and - whole ingredient >$before< containing 'and' is unknown ingredient")
+					if $log->is_debug();
+
 				# otherwise check the 2 sub ingredients
 				my $canon_ingredient1 = canonicalize_taxonomy_tag($ingredients_lc, "ingredients", $ingredient1);
 				my $canon_ingredient2 = canonicalize_taxonomy_tag($ingredients_lc, "ingredients", $ingredient2);
 
-				$debug_ingredients and $log->debug(
-					"ingredient containing 'and' did not exist. 2 known ingredients?",
-					{
-						before => $before,
-						canon_ingredient => $canon_ingredient,
-						canon_ingredient1 => $canon_ingredient1,
-						canon_ingredient2 => $canon_ingredient2
-					}
-				) if $log->is_debug();
-
+				# both ingredients are in the taxonomy as is
 				if (    (exists_taxonomy_tag("ingredients", $canon_ingredient1))
 					and (exists_taxonomy_tag("ingredients", $canon_ingredient2)))
 				{
+					$debug_ingredients
```

Comment:
This works very well, but I think we could further simplify it:

We change the parse_processing_from_text in a function that first check if the ingredient exists, and otherwise tries to parse the processing.

Then we call it on both ingredients right away, and we can just check if both ingredients were recognized (either as-is, or by removing processing) in only one condition.

And that way we would also recognize "fried steak and grilled mushrooms" with that same condition.

Reply:
I tried something in that direction